### PR TITLE
Add option to insert blanks without cutting input

### PR DESCRIPTION
**Pull request by ciebson123**
# Description
todo

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [x] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: research/blanks/argparse.py [hunk 1]
```diff
@@ -199,5 +199,8 @@ def introduce_parser_arguments(
     parser.add_argument("--blanks_use_multiple_embeddings", action="store_true")
     parser.add_argument("--blanks_use_custom_attention", action="store_true")
     parser.add_argument("--blanks_use_only_last_blank_loss", action="store_true")
+    parser.add_argument(
+        "--blanks_insert_blanks_without_replacement", action="store_true"
+    )
```
Comment (maciek-pioro) [inline research/blanks/argparse.py:204]: `--extend_sequence_by_n_blanks` and so on in the rest of the PR

File: research/blanks/packer.py [hunk 2]
```diff
@@ -37,13 +38,21 @@ def __init__(
...
         input_tokens = sample.input_ids
         target_tokens = sample.target_ids
         seq_len = len(input_tokens)
+        if self.insert_blanks_without_replacement:
+            input_tokens.extend(
+                [0] * self.n_blanks
+            )  # will be cut off by inserting blanks
+            target_tokens.extend([0] * self.n_blanks)
+            seq_len += self.n_blanks
+            sample.should_calculate_loss.extend([1] * self.n_blanks)
```
Comment (maciek-pioro) [inline research/blanks/packer.py:55]: Maybe `sample.should_calculate_loss.extend([-1] * self.n_blanks)` and put an assert at the end of the function checking that there is no -1's.

**Review by maciek-pioro — COMMENTED** [review]
Some changes needed, logic seems impeccable.

File: research/blanks/packer.py (hunk 2 above)
Comment (ciebson123) [reply research/blanks/packer.py:55]: Not how it works, this is either copied without any change or made from grouds up. We already have an assert should_calculate_loss == [1] * seq_len

File: research/blanks/argparse.py (hunk 1 above)
Comment (ciebson123) [reply research/blanks/argparse.py]: Done

**Review by maciek-pioro — APPROVED** [review]

**Pull request merged**